### DISCUSSION TITLE
How to create a new blog project under website project?

### DISCUSSION THREAD

**Discussion by e-alizadeh** [quarto-dev/quarto-cli#2389] [Q&A] [answered]
I'm new to quarto. I want to create a blog under my main website. So, does it make sense to create two quarto projects?
So for this, I can first run `quarto create-project mysite --type website` and then under `mysite` directory I run `quarto create-project myblog --type website:blog`. This way the directory `mysite/myblog` would contain the blog posts.

Does this make sense? This way I'm going to have two `_quarto.yml` along with other config files.

**Comment by mcanouil** [comment, marked as answer]
Why not simply create a website/blog post and try to nest Quarto projects? 🤔
Take a look at <https://quarto.org>, it's a website and it has a blog.

**Reply by e-alizadeh** [reply]
Thanks @mcanouil for your message. 
That was actually my point. To first create a website project and then create a blog project under it. But if you look at https://github.com/quarto-dev/quarto-web/ you would see that it only contains one `_quarto.yml` file and the project type is `website`. I don't see any other `_quarto.yml` file. I just thought of asking this before I start migrating my website and to save some time if there is any suggestion. 
Thanks again.

**Reply by mcanouil** [reply]
I still don't understand why build a website within a website, which will require two rendering step while a website can include a blog (which is just a page and a "listing"). The type of a blog is `type: website`.

Why do you want to have multiple configuration files for the same thing?

`quarto create-project myblog --type website:blog` will simply help you to have the blog structure in addition to the regular stuff for a Quarto website.

**Reply by dragonstyle** [reply]
When you run `quarto create-project` for a blog, Quarto is really just creating a website which is configured as a blog (with a particular navigation structure and the home page set to contain a listing). To add a blog to your current website project, I think it should be as simple as creating a page for your blog (e.g. `blog.qmd`) with a listing pointing at a posts directory. Here is a sample `blog.qmd`:

```markdown
---
title: "blog"
listing:
  contents: posts
  sort: "date desc"
  type: default
  categories: true
  sort-ui: false
  filter-ui: false
page-layout: full
title-block-banner: true
---
```

From there you make a posts directory, in it I would create a `_metadata.yml` file with the following:

```markdown
# options specified here will apply to all posts in this folder

# freeze computational output
# (see https://quarto.org/docs/projects/code-execution.html#freeze)
freeze: true

# Enable banner style title blocks
title-block-banner: true
```

which will configure these settings for any posts in the folder. 

Now you've got the basics set up, you just need to start adding `.qmd` files in the posts folder. 

HTH!

**Reply by e-alizadeh** [reply]
Thanks @dragonstyle for the example. It now makes more sense to me.

**Reply by mcanouil** [reply]
@e-alizadeh Maybe mark the thread as the answer, if your question has been answered fully, to let others know it has been answered. 😉
